src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLogger.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.Shared.Pools;
+
+namespace Microsoft.Extensions.Telemetry.Logging;
+
+#pragma warning disable CA1031
+
+// NOTE: This implementation uses thread local storage. As a result, it will fail if formatter code, enricher code, or
+//       redsctor code calls recursively back into the logger. Don't do that.
+//
+// NOTE: Unlike the original logger in dotnet/runtime, this logger eats exceptions thrown from invoked loggers, enrichers,
+//       and redactors, rather than forwarding the exceptions to the caller. The fact an exception occured is recorded in
+//       the event log instead. The idea is that failures in the telemetry stack should not lead to failures in the
+//       application. It's better to keep running with missing telemetry rather than crashing the process completely.
+
+internal sealed partial class ExtendedLogger : ILogger
+{
+    private const string ExceptionStackTrace = "stackTrace";
+
+    private readonly ExtendedLoggerFactory _factory;
+
+    public LoggerInformation[] Loggers { get; set; }
+    public MessageLogger[] MessageLoggers { get; set; } = Array.Empty<MessageLogger>();
+    public ScopeLogger[] ScopeLoggers { get; set; } = Array.Empty<ScopeLogger>();
+
+    public ExtendedLogger(ExtendedLoggerFactory factory, LoggerInformation[] loggers)
+    {
+        _factory = factory;
+        Loggers = loggers;
+    }
+
+    public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
```

Comment:
How does it call into the OpenTelemetry processors here? It will require us to somehow create the LogRecord object and then pass it to the OpenTelemetry processors to further process.

Reply:
This is the top-level logger implementation, it delegates to all low-level loggers. In a normal scenario, there will be an Otel-specific logger register which will get called with a TState that has all the enriched properties and redacted state.

Reply:
So basically:

* Application layer calls ILogger.Log using a LoggerMessageState object as the TState.
* ExtendedLogger.Log is invoked and recognizes the LoggerMessageState type. It performs enrichment and redaction, resulting in a ModernPropertyJoiner object.
* ExtendedLogger.Log then calls all lower-level loggers, passing the ModernPropertyJoiner object as TState. Thus, all lower-level loggers receives enriched and redacted state.

Reply:
Got it. So, we have already done everything and then one can add OpenTelemetry logger and that logger will directly receive the logs which are alreayd redacted and enriched.